inst/templates/ad_advs.R

```diff
@@ -44,8 +44,17 @@ bmi_for_age <- who_bmi_for_age_boys %>%
         SEX == 2 ~ "F",
         TRUE ~ NA_character_
       ),
-      AGE = AGE * 30.4375
-    )) %>%
+      # Ensure first that Age unit is "DAYS"
+      AGE = round(AGE * 30.4375)
+    ) %>%
+    # Interpolate the AGE by SEX
+    derive_interp_records(
+      by_vars = exprs(SEX),
+      parameter = "BMI"
+    ) %>%
+    # Keep patients >= 2 yrs till 20 yrs - Remove duplicates for 730 Days old which
+    # must come from WHO metadata only
+    filter(AGE >= 730.5 & AGE <= 7305)) %>%
```

Comment:
do we need the `& AGE <= 7305` here? seems redundant to me, and same for other lines

Reply:
This is because `cdc_bmiage` metadata includes data till 240.5 months ( = 7320.22 days which is 20.04 yrs old). Should it be considered ? When David interpolated the data, he fixed the upper limit to 7305 days.

Reply:
ok np, fine by me then and you do explain in the comment that its only up to 20 years